cli/src/main/java/com/devonfw/tools/ide/util/MapUtil.java

```diff
@@ -0,0 +1,34 @@
+package com.devonfw.tools.ide.util;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility class for operations with maps.
+ */
+public class MapUtil {
+
+  /**
+   * Creates a {@link HashMap} with the given {@code keys} and {@code values} which are passed as {@link List lists}.
+   * The map is populated by iterating through both lists simultaneously until one of the list is exhausted.
+   */
+  public static <K, V> Map<K, V> createMapfromLists(List<K> keys, List<V> values) {
+
+    Map<K, V> resultMap = new HashMap<>();
+
+    // Create iterators for both lists
+    Iterator<K> keysIterator = keys.iterator();
+    Iterator<V> valuesIterator = values.iterator();
+
+    // Iterate through both iterators simultaneously
+    while (keysIterator.hasNext() && valuesIterator.hasNext()) {
+      K key = keysIterator.next();
+      V value = valuesIterator.next();
+      resultMap.put(key, value);
+    }
+
+    return resultMap;
+  }
+}
```

Comment:
This class is never used so why adding it?

Reply:
OK, github collapsed the diff of `BuildSecurityJsonFiles` where it is used. Still I would question the need for this. It can also be done directly in `BuildSecurityJsonFiles` by using the Java `Stream` API (`Streams.zip(keys.stream(), values.stream(), Maps::immutableEntry).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))`). Also if we have it here it implies the need to write JUnits for it and generate coverage, etc.

Reply:
> Streams.zip(keys.stream(), values.stream(), Maps::immutableEntry).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))

This is a nice improvement. But it sadly didn't occur to me.